taichi/backends/vulkan/runtime.cpp

```diff
@@ -61,16 +53,16 @@ class StopWatch {
 };
 
 using BufferEnum = TaskAttributes::Buffers;
-using InputBuffersMap = std::unordered_map<BufferEnum, VkBufferWithMemory *>;
+using InputBuffersMap = std::unordered_map<BufferEnum, DeviceAllocation *>;
 
```

Comment:
Left as a TODO since it's causing quite some trouble rn as the context_buffers are in unique ptrs which can be null if the kernel doesn't use args, and the code gets a bit messy since de-refercing a null is definitely not good but otherwise there need to be if statements to pass in a null allocation depending on the unique pointer's "null-ness"